### PR TITLE
TypeScript definitions update: 'strictNullCheck' and 'onXxxCapture' events

### PR DESCRIPTION
**Pull request by arusanov**
1. Added strict null checks for typings
2. Changed `class` and `className` types to `string` (preact 8 changes)
3. Added most common `capture` event typings that were constants in preact 7
4. Added `onReset` event
5. Removed `linkState`

### CODE REVIEW

**Comment by arusanov** [comment]
@robertknight @developit ?

**Comment by Hotell** [comment]
strict null checks and more was already submitted https://github.com/developit/preact/pull/621

**Comment by arusanov** [comment]
Anyway this PR contains type changes for `preact 8.x` and some missing event like `onLoad`, `onResize` etc

**Comment by Hotell** [comment]
yup I know

File: src/vdom/component.js [hunk 1]
```diff
@@ -144,6 +144,9 @@ export function renderComponent(component, opts, mountAll, isChild) {
 		if (initialBase && base!==initialBase && inst!==initialChildComponent) {
 			let baseParent = initialBase.parentNode;
 			if (baseParent && base!==baseParent) {
+				if (toUnmount) {
```
Comment (marvinhagemeister) [inline src/vdom/component.js]: I'd really love to merge this PR but I'm a bit hesitant because of this change here, because this changes the behaviour of preact itself instead of only updating the TypeScript types. 

@arusanov Can you split this change here into its own PR?

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "preact",
-  "version": "8.2.7",
+  "version": "8.2.8",
```
Comment (marvinhagemeister) [inline package.json]: This PR should only change the TypeScript typings.

File: package.json [hunk 3]
```diff
@@ -34,7 +34,7 @@
     "test:size": "bundlesize",
     "lint": "eslint debug devtools src test",
     "prepublish": "npm run build",
-    "smart-release": "npm run build && npm test && git commit -am $npm_package_version && git tag $npm_package_version && git push && git push --tags && npm publish",
+    "smart-release": "npm run build && npm test",
```
Comment (marvinhagemeister) [inline package.json]: Not sure if this change is intended

File: src/preact.d.ts [hunk 4]
```diff
@@ -37,10 +37,11 @@ declare namespace preact {
 		shouldComponentUpdate?(nextProps:PropsType,nextState:StateType,nextContext:any):boolean;
 		componentWillUpdate?(nextProps:PropsType,nextState:StateType,nextContext:any):void;
 		componentDidUpdate?(previousProps:PropsType,previousState:StateType,previousContext:any):void;
+		componentWillReplace?():void;
```
Comment (marvinhagemeister) [inline src/preact.d.ts]: I can't find any reference to `componentWillReplace` in our source code or `preact-compat`. Perhaps I'm missing something?

File: src/preact.d.ts [hunk 5]
```diff
@@ -564,8 +573,8 @@ declare namespace JSX {
 		charSet?:string;
 		challenge?:string;
 		checked?:boolean;
-		class?:string | { [key:string]: boolean };
-		className?:string | { [key:string]: boolean };
+		class?:string;
```
Comment (marvinhagemeister) [inline src/preact.d.ts]: Awesome! Thanks for spotting this! 🎉

**Comment by arusanov** [comment]
I guess @marvinhagemeister can close this as more better typings was already merged

**Comment by marvinhagemeister** [comment]
@arusanov I see, would love to get the `class/className` change in master, though :)
